app/controllers/dashboards_controller.rb

```diff
@@ -1,9 +1,15 @@
 class DashboardsController < ApplicationController
   def show
-    @environment = Environment.find(params[:id])
-    @deploys = Project.alphabetical.with_deploy_groups.each_with_object({}) do |project, hash|
+    @environment = Environment.find_by_name!(params[:id])
+    @deploys = ordered_projects.each_with_object({}) do |project, hash|
       hash[project] = project.last_deploy_by_group
       hash[project].select! { |id, _v| @environment.deploy_group_ids.include?(id) }
     end
   end
+
+  private
+
+  def ordered_projects
+    (current_user.starred_projects.alphabetical.with_deploy_groups + Project.alphabetical.with_deploy_groups).uniq
```

Comment:
that looks like it was fun to build ;)

On Thu, Mar 12, 2015 at 7:56 AM, Shane Hender <notifications@github.com>
wrote:

> In app/controllers/dashboards_controller.rb
> <https://github.com/zendesk/samson/pull/288#discussion_r26308460>:
>
> >        hash[project] = project.last_deploy_by_group
> >        hash[project].select! { |id, _v| @environment.deploy_group_ids.include?(id) }
> >      end
> >    end
> > +
> > +  private
> > +
> > +  def ordered_projects
> > +    (current_user.starred_projects.alphabetical.with_deploy_groups + Project.alphabetical.with_deploy_groups).uniq
>
> Got the join down to this, kinda ugly, but it was complex.
>
> SELECT Projects.*, count(stars.id) as star_count FROM "projects" left join stars on stars.user_id = 1 and stars.project_id = projects.id GROUP BY projects.id  ORDER BY star_count DESC, projects.name DESC
>
>   scope :ordered_for_user, ->(user) {
>     select('Projects.*, count(stars.id) as star_count').
>       joins("left outer join stars on stars.user_id = #{sanitize(user.id)} and stars.project_id = projects.id").
>       group('projects.id').
>       order('star_count desc').
>       alphabetical
>   }
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/zendesk/samson/pull/288/files#r26308460>.
>